libnmstate/state.py

```diff
@@ -474,6 +475,11 @@ def _capitalize_mac(self):
             if mac:
                 ifstate[Interface.MAC] = mac.upper()
 
+    def _remove_empty_description(self):
+        for ifstate in six.viewvalues(self.interfaces):
+            if not ifstate.get(Interface.DESCRIPTION):
+                ifstate.pop(Interface.DESCRIPTION, None)
```

Comment:
A non-must one: The first asks if there is a key or it is empty, the second cannot trust the first so it needs to check again.
This is a direct approach:
```
if ifstate.get(Interface.DESCRIPTION) == ''
    del ifstate[Interface.DESCRIPTION]
```